cmp/grid/GridModel.js

```diff
@@ -94,6 +94,8 @@ export class GridModel {
     @observable.ref columns = [];
     /** @member {ColumnState[]} */
     @observable.ref columnState = [];
+    /** @member {Object} */
+    @observable.ref expandState = {};
```

Comment:
What's our contract around keeping this expand state accurate and in sync?  I see we call on `expand/collapseAll()` + `onRowGroupOpened` in the Grid. But what about initial setup, or on a single row collapse?

This is another case where performance might be an issue (especially if the above comment would cause us to call more frequently).  Now we're triggering a pretty intensive iteration over the nodes on these common actions. Realize this was added for a reason 😄 but want to ensure we're being as efficient as possible.

Reply:
`onRowGroupOpened` also triggers on collapse (a bit of a misnomer). I've also since added code to update expandState on data change, since that could lead to stale expandState. The data change reaction should cover the initial setup